thefuck/rules/dnf_install.py

```diff
@@ -0,0 +1,22 @@
+
+
+def get_packages(command):
+    splitted_command = command.script.split()
+    if splitted_command[0] == 'sudo':
+        return splitted_command[3:]
+    return splitted_command[2:]
+
+
+def match(command):
+    typos = ('dfn install', 'dnf istall', 'dfn istall')
```

Comment:
This is not very generic:
  * The case `dnf` vs. `dfn` vs. any other similar typo is already taken care of by the `no_command` rule;
  * The case of `istall` vs. `install` should be taken care of with `utils.get_closest` and a list of possible candidates (I guess there is more to `dnf` than just an `install` subcommand).